scripts/Merge_Trajectories.py

```diff
@@ -0,0 +1,190 @@
+                if last_frame == not_yet_found:
+                    last_frame = current_frame
+                elif last_frame != current_frame and last_frame + 1 != current_frame:
+                    raise IncorrectTrajectoryException(
+                        f"file {file} has missing frames ~ missed frame {last_frame+1}")
+                else:
+                    last_frame = current_frame
+    return True
+
+
+def startingFrame(file):
+    with open(file, "r") as opened_file:
+        for line in opened_file:
+            if line != "\n" and not line.startswith("#"):
+                return FramefromLine(line)
+
+
+def endingFrame(file):
+    with open(file, "r") as opened_file:
+        lines = opened_file.readlines()
+        lines.reverse()
+
+    for line in lines:
+        if line != "\n" and not line.startswith("#"):
+            return FramefromLine(line)
+
+
+def checkdata(files):
+    """checks if files are matching one Trajectory"""
+    end_frames = []
+    start_frames = []
+    for file in files:
+        if not isComplete(file):
+            return False
+        start_frames.append(startingFrame(file))
+        end_frames.append(endingFrame(file))
+    for start_frame in start_frames:
+        # a starting frame is considered valid if it is the first frame of the given trajectories
+        # a starting frame is considered valid if it comes after an end frame
+        if start_frame != min(start_frames) and start_frame - 1 not in end_frames:
+            raise IncorrectTrajectoryException(f"the starting frame {start_frame} is not matching the other files")
+        # a starting frame is invalid if it appears more than once
+        if start_frames.count(start_frame) > 1:
+            raise IncorrectTrajectoryException(f"multiple files start with the frame {start_frame}")
+    return True
+
+
+def addDatas(trajecs, output, debug):
+    """will merge all data into the output-file
+      "trajecs" is a list of all Trajectory files"""
+    files = trajecs
+
+    # creates a list with all first Frames
+    first_frames = []
+    for file in files:
+        first_frames.append(startingFrame(file))
+
+    debug_counter = 1
+    debug_length = len(first_frames)
+    # until all files are written
+    while first_frames:
+        # determines the lowest starting frame
+        next_frame = min(first_frames)
+        for i, frame in enumerate(first_frames):
+            # adds all data from that file to the output file
+            if frame == next_frame:
+                if debug:
+                    log.debug(f"writing file {debug_counter}/{debug_length}")
+                write(files[i], output)
+                # removes file and the first line from the lists after the data has been written
+                files.pop(i)
+                first_frames.pop(i)
+                debug_counter += 1
+                break
+
+    log.info(f'new File "{output}" was created.')
+
+
+def write(inputfile, output):
```

Comment:
The name `write` does not indicate that repeated calls to this function will append to output, so a better name would be for example: `append_to_output` or `append_frames_to_output`

Reply:
Now called append_to_output